.github/workflows/test.yml

```diff
@@ -18,60 +18,43 @@ env:
   PARALLEL: true
 
 jobs:
+  # separate job for parallelism
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Verify
+      uses: golangci/golangci-lint-action@v2
+      with:
+        version: v1.33.2
+        working-directory: go-controller
+        args: --modules-download-mode=vendor --timeout=15m0s
+
   build:
     name: Build
     runs-on: ubuntu-latest
     steps:
     - name: Set up Go
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v2
       with:
         go-version: ${{ env.GO_VERSION }}
       id: go
 
     - name: Check out code into the Go module directory
       uses: actions/checkout@v2
 
-    - name: Set up environment
-      run: |
-        export GOPATH=$(go env GOPATH)
-        echo "GOPATH=$GOPATH" >> $GITHUB_ENV
-        echo "$GOPATH/bin" >> $GITHUB_PATH
-
-    - name: Verify
-      run: |
-        pushd go-controller
-          make gofmt
```

Comment:
@squeed what runs gofmt now? golangci-lint doesn't appear to run gofmt by default according to https://golangci-lint.run/usage/linters/

Reply:
@dcbw it's enabled already in the .golangci.yml

```
$ make lint
level=info msg="[config_reader] Config search paths: [./ /app / /root]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 11 linters: [deadcode errcheck gofmt gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck]"
```

Reply:
@squeed yeah, but you're not running 'make lint' anywhere in here. Do you mean that `golangci/golangci-lint-action@v2` will pick up .golangci.yml automatically?

Reply:
All I see in your CI run is:

```
* run golangci-lint
  Running [/home/runner/golangci-lint-1.33.2-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=go-controller --modules-download-mode=vendor --timeout=15m0s] in [/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller] ...
  golangci-lint found no issues
  Ran golangci-lint in 2067ms
```

Reply:
good point. Added `--verbose`, let's see if that makes a difference.

Reply:
@dcbw Yup, we still run `gofmt`.

Reply:
BTW, I switched to the action because it's more aggressive about caching and faster in general - 2m45 vs 3m30.